### PR TITLE
Fix Leksah

### PR DESCRIPTION
**Pull request by abbradar**
This adds Leksah wrapper and adds propagation of gtk3 dependency to gtksourceview (to skip adding it to auto-generated Hackage packages where needed). This also shows fixes needed in `cabal2nix` for some packages to build -- cc @peti . `process-leksah` is removed from dependencies because it is actually needed only for GHC <= 7.2 (maybe we want to make it `null` in later GHCs?).

Not for merging until `cabal2nix` fixes are there!

### CODE REVIEW

**Comment by cstrahan** [comment]
Speaking of fixes needed in `cabal2nix`, it appears `cabal2nix` doesn't remove the bounds for dependencies listed under a conditional branch (`haskell-src-meta` is an example). @abbradar, do you know if there's an open issue on `cabal2nix`'s repo? Or should I open one?

**Comment by abbradar** [comment]
@cstrahan Not a one I've heard of (and nothing from trivial search queries) -- be the first one!

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -53215,7 +53215,7 @@ self: {
        description = "Binding to the GtkSourceView library";
        license = stdenv.lib.licenses.lgpl21;
        hydraPlatforms = stdenv.lib.platforms.none;
-     }) { inherit (pkgs.gnome) gtksourceview;};
+     }) { inherit (pkgs.gnome3) gtksourceview;};
```
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: This should probably be an `override` in the common configuration.

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 2]
```diff
@@ -124381,15 +124381,15 @@ self: {
...
        buildDepends = [ base bytestring cairo glib gtk3 mtl pango text ];
        buildTools = [ gtk2hs-buildtools ];
-       pkgconfigDepends = [ webkit ];
+       pkgconfigDepends = [ webkitgtk24x ];
        homepage = "http://projects.haskell.org/gtk2hs/";
        description = "Binding to the Webkit library";
        license = stdenv.lib.licenses.lgpl21;
        hydraPlatforms = stdenv.lib.platforms.none;
-     }) { inherit (pkgs) webkit;};
+     }) { inherit (pkgs) webkitgtk24x;};
```
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Same here; please override `webkit` with the proper name in the common configuration.

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 3]
```diff
@@ -124398,11 +124398,11 @@ self: {
        sha256 = "0w71phbs11iim2gb89bxix0grx9k7fbvabzjjsw7l24wwgfn8ad6";
        buildDepends = [ base glib gtk3 webkitgtk3 ];
        buildTools = [ gtk2hs-buildtools ];
-       pkgconfigDepends = [ webkit ];
+       pkgconfigDepends = [ webkitgtk24x ];
        description = "JavaScriptCore FFI from webkitgtk";
        license = stdenv.lib.licenses.bsd3;
        hydraPlatforms = stdenv.lib.platforms.none;
-     }) { inherit (pkgs) webkit;};
+     }) { inherit (pkgs) webkitgtk24x;};
```
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Dito.

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 4]
```diff
@@ -76433,7 +76433,7 @@ self: {
      , binary-shared, bytestring, Cabal, conduit, conduit-extra
      , containers, deepseq, directory, executable-path, filepath, ghc
      , haddock-api, hslogger, HTTP, HUnit, ltk, network, network-uri
-     , parsec, pretty, process, process-leksah, resourcet, strict, text
+     , parsec, pretty, process, resourcet, strict, text
```
Comment (peti) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Fixed in cabal2nix. Thanks for pointing this out!

File: pkgs/development/haskell-modules/hackage-packages.nix (hunk 1 above)
Comment (abbradar) [inline pkgs/development/haskell-modules/hackage-packages.nix]: Oh, right -- I've forgotten I can just override there. Thanks!

**Comment by peti** [comment]
Will there be an update to this PR to add the overrides instead of patching `hackage-packages.nix`? It would be great to have Leksah compiling again!

**Comment by abbradar** [comment]
I'm compiling it right now -- we need bigger changes than I anticipated, namely old versions of `haddock-api` for GHC 7.8 and `jsaddle` workarounds. Speaking of with, would you mind of me adding `configuration-ghc.nix` which would contain common fixes for GHC? This is needed because `jsaddle` has different dependencies based on its `ghcjs` flag (enabled by default) -- I want to override it and change dependencies for all GHCs but not for GHCJS.

**Comment by peti** [comment]
For the time being, can you make the `jsaddle` changes in one of the existing `configuration-ghc-X.Y.nix` files, please? I'm wary of adding yet another level of overriding to the package set.

**Comment by abbradar** [comment]
OK, I'll add it to 7.8-7.10-HEAD then -- more if somebody would ask.

**Comment by abbradar** [comment]
Done, also needs https://github.com/NixOS/cabal2nix/pull/157 now.

**Comment by abbradar** [comment]
After some thinking I've reworked the wrapper, removing `packages` input -- it seems to be not needed.

**Comment by peti** [comment]
Test builds for the updated package set with older versions of Haddock available run at http://hydra.cryp.to/jobset/nixpkgs/haskell-updates. I'll merge those to master ASAP.

**Comment by abbradar** [comment]
Rebased and pushed obvious parts to master. Can we push the rest?

BTW, @peti, do you mind me pushing straightforward patches like https://github.com/NixOS/nixpkgs/commit/08bf6c93c52aad248b3e0807ad890c100d90ebf3 and https://github.com/NixOS/nixpkgs/commit/3956bc8e6440c3e97ec9d9a08bf06792b43f801d directly (and maybe merging PRs like this), or would you prefer getting a look first?

**Comment by pSub** [comment]
Any updates on this? I've tried to start it, but it fails with 

```
leksah: cabal: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

**Comment by peti** [comment]
@abbradar, please feel free to merge any patch you feel comfortable with. IMHO, pull requests are great to get feedback from someone else, but it's by no means necessary to go through a PR every time. Doing that would just slow down development.

**Comment by peti** [comment]
I merged the patch manually in 0e223ff (after some trivial editing). @cstrahan, I hope I didn't break anything with regard to ghcjs.

**Comment by abbradar** [comment]
@peti, Good, will do.
@pSub Strange, looks like leksah also needs `cabal-install` in his PATH. I can add it to the wrapper later this evening -- if you want it earlier, please add it to the package list (https://github.com/NixOS/nixpkgs/commit/77a153a6552f08766d7d3b2747e5c70ca3b99879#diff-9f5c5eb55340204046efa603d4b32c25R4) and push the patch.

**Pull request closed without merging**